File: 2-Basic-Mathematical-Knowledge-for-Algorithms/007-Number_of_Multiples_1/java/Main.java
import java.util.Scanner;

public class Main {
    static long gcd(long a, long b) {
        while (b != 0) {
            long temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    static long lcm(long a, long b) {
        return (a / gcd(a, b)) * b;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        // 1. 入力処理
        long n = sc.nextLong();
        long x = sc.nextLong();
        long y = sc.nextLong();

        // 2. LCM計算
        long lcmXY = lcm(x, y);

        // 3. 包除原理適用
        long result = n / x + n / y - n / lcmXY;

        // 4. 出力処理
        System.out.println(result);
        
        sc.close();
    }
}

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/007-Number_of_Multiples_1/rust/src/main.rs
use std::io;

fn gcd(mut a: u64, mut b: u64) -> u64 {
    while b != 0 {
        let temp = b;
        b = a % b;
        a = temp;
    }
    a
}

fn lcm(a: u64, b: u64) -> u64 {
    (a / gcd(a, b)) * b
}

fn main() {
    // 1. 入力処理
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let numbers: Vec<u64> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Please enter a number"))
        .collect();
    
    let n = numbers[0];
    let x = numbers[1];
    let y = numbers[2];

    // 2. LCM計算
    let lcm_xy = lcm(x, y);

    // 3. 包除原理適用
    let result = n / x + n / y - n / lcm_xy;

    // 4. 出力処理
    println!("{}", result);
}